routers/web/repo/milestone.go

```diff
@@ -174,6 +265,76 @@ func EditMilestone(ctx *context.Context) {
 	}
 	ctx.Data["title"] = m.Name
 	ctx.Data["content"] = m.Content
+
+	var selectLabelIDs []int64
+	var labelExclusiveScopes []string
+	selectLabels := ctx.FormString("labels")
+	labels, err := issues_model.GetLabelsByRepoID(ctx, ctx.Repo.Repository.ID, "", db.ListOptions{})
+	if err != nil {
+		ctx.ServerError("GetLabelsByRepoID", err)
+		return
+	}
+	if len(selectLabels) > 0 && selectLabels != "0" {
+		selectLabelIDs, err = base.StringsToInt64s(strings.Split(selectLabels, ","))
+		if err != nil {
+			ctx.ServerError("StringsToInt64s", err)
+			return
+		}
+		// Get the exclusive scope for every label ID
+		labelExclusiveScopes = make([]string, 0, len(selectLabelIDs))
+		for _, labelID := range selectLabelIDs {
+			foundExclusiveScope := false
+			for _, label := range labels {
+				if label.ID == labelID || label.ID == -labelID {
+					labelExclusiveScopes = append(labelExclusiveScopes, label.ExclusiveScope())
+					foundExclusiveScope = true
+					break
+				}
+			}
+			if !foundExclusiveScope {
+				labelExclusiveScopes = append(labelExclusiveScopes, "")
+			}
+		}
+	}
+	hasSelected := len(selectLabelIDs) > 0
```

Comment:
I guess I have seen such code not only once  😁 


could there be a shared function?